lazy-either.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const LazyEither = require('lazy-either');
+const R = require('ramda');
+const S = require('sanctuary');
+
+const lift = LazyEither.lift;
+
+// join :: String -> String -> String
+const join = R.curryN(2, path.join);
+
+// readFile :: String -> String -> LazyEither (Either e String)
+const readFile = R.curry((encoding, filename) =>
+  new LazyEither(resolve => {
+    fs.readFile(filename, {encoding: encoding}, (err, data) =>
+        resolve(err ? S.Left(err) : S.Right(data.toString())));
+  }));
+
+// readFiles :: [String] -> LazyEither (Either e String)
+const readFiles = files => R.traverse(LazyEither.of, readFile('utf8'), files);
```

Comment:
This is unnecessary function wrapping. We could partially apply `R.traverse` and assign the result to `readFiles`.